tests/integration/tests/conftest.py

```diff
@@ -100,7 +100,40 @@ def instances(
 
     yield instances
 
-    _harness_clean(h)
+    if not config.SKIP_CLEANUP:
+        # Cleanup after each test.
+        # We cannot execute _harness_clean() here as this would also
+        # remove the session_instance. The harness ensures that everything is cleaned up
+        # at the end of the test session.
+        for instance in instances:
+            h.delete_instance(instance.id)
```

Comment:
the double negative really broke my brain for a bit.  

```suggestion
    if config.SKIP_CLEANUP:
        LOG.warning("Skipping clean-up of instances, delete them on your own"
        return
    # Cleanup after each test.
    # We cannot execute _harness_clean() here as this would also
    # remove the session_instance. The harness ensures that everything is cleaned up
    # at the end of the test session.
    for instance in instances:
        h.delete_instance(instance.id)
```

Reply:
is `_harness_clean` no longer used?  Can it be deleted

Reply:
Thanks, removed the double-negative.

`_harness_clean` is still used when the harness is cleaned up at the end of the session. https://github.com/canonical/k8s-snap/blob/63498cfc1c7ba75cf15ce9d5db3adb94df09b316/tests/integration/tests/conftest.py#L46

It is only removed from the `instances` as this would also cleanup the session instance.